src/lib/OS/SimOS.cc

```diff
@@ -90,18 +93,21 @@ void SimOS::tick() {
           // scheduled (i.e. on first tick of simulation)
           if (currContext.TID != -1) {
             // Find the corresponding process in map
-            auto currProc = processes_.find(currContext.TID)->second;
-            assert(
-                (currProc->status_ == procStatus::executing) &&
-                "[SimEng:SimOS] Process updated when not in executing state.");
-            // Only update values which have changed
-            currProc->context_.pc = currContext.pc;
-            currProc->context_.regFile = currContext.regFile;
-            // Change status from Executing to Waiting
-            currProc->status_ = procStatus::waiting;
-            waitingProcs_.push(currProc);
+            auto currProc = processes_.find(currContext.TID);
```

Comment:
Nitpick: Could do `auto itr = processes_.find()` and then `auto process = itr->second`

Reply:
Done